File: old/actions/install_openssh_server.py
"""
Install openssh-server action
"""
from .base import Action
from cli import Apt

class InstallOpensshServerAction(Action):
    """Install openssh-server package"""
    description = "openssh-server installation"

    def execute(self) -> bool:
        """Install openssh-server"""
        install_cmd = Apt().install(["openssh-server"])
        output, exit_code = self.pct_service.execute(self.container_id, install_cmd)
        if exit_code is not None and exit_code != 0:
            self.logger.error("Failed to install openssh-server: %s", output)
            return False
        return True

File: src/actions/install_openssh_server.go
package actions

import (
	"strconv"
	"enva/cli"
	"enva/libs"
	"enva/services"
)

// InstallOpensshServerAction installs openssh-server package
type InstallOpensshServerAction struct {
	*BaseAction
}

func NewInstallOpensshServerAction(sshService *services.SSHService, aptService *services.APTService, pctService *services.PCTService, containerID *string, cfg *libs.LabConfig, containerCfg *libs.ContainerConfig) Action {
	return &InstallOpensshServerAction{
		BaseAction: &BaseAction{
			SSHService:   sshService,
			APTService:   aptService,
			PCTService:   pctService,
			ContainerID: containerID,
			Cfg:          cfg,
			ContainerCfg: containerCfg,
		},
	}
}

func (a *InstallOpensshServerAction) Description() string {
	return "openssh-server installation"
}

func (a *InstallOpensshServerAction) Execute() bool {
	containerIDInt, err := strconv.Atoi(*a.ContainerID)
	if err != nil {
		libs.GetLogger("install_openssh_server").Printf("Invalid container ID: %s", *a.ContainerID)
		return false
	}
	installCmd := cli.NewApt().Install([]string{"openssh-server"})
	output, exitCode := a.PCTService.Execute(containerIDInt, installCmd, nil)
	if exitCode != nil && *exitCode != 0 {
		libs.GetLogger("install_openssh_server").Printf("Failed to install openssh-server: %s", output)
		return false
	}
	return true
}

